src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4288,14 +4288,181 @@ private static bool UseLegacyWarnings(BoundExpression expr)
 
         public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
         {
-            // https://github.com/dotnet/roslyn/issues/29618: Assign each of the deconstructed values,
-            // and handle deconstruction conversion for node.Right.
-            VisitLvalue(node.Left);
-            VisitRvalue(node.Right.Operand);
+            var left = node.Left;
+            var right = node.Right;
+            var variables = GetDeconstructionAssignmentVariables(left);
```

Comment:
>GetDeconstructionAssignmentVariables [](start = 28, length = 36)

Consider moving the `GetDeconstructionAssignmentVariables` and freeing of those variables into the `else`, since unused in the first branch of the `if`

Reply:
Currently, `GetDeconstructionAssignmentVariables` has side-effects in that it calls `VisitLvalue()` on each variable, so we should execute that method regardless.

---
In reply to: [252773231](https://github.com/dotnet/roslyn/pull/32994#discussion_r252773231) [](ancestors = 252773231)